web/src/message_edit.ts

```diff
@@ -739,20 +798,22 @@ export function toggle_resolve_topic(
             }
 
             if (xhr.responseJSON) {
-                if (xhr.responseJSON.code === "MOVE_MESSAGES_TIME_LIMIT_EXCEEDED") {
+                const {code} = z.object({code: z.string()}).parse(xhr.responseJSON);
```

Comment:
Error responses need to be parsed using `safeParse`, because we might have gotten an HTML error from nginx instead of a JSON error from Django ([previous discussion](https://chat.zulip.org/#narrow/stream/6-frontend/topic/typescript.20migration/near/1794568)).